pkgpanda/__init__.py

```diff
@@ -652,6 +674,13 @@ def get_active(self):
         ids = set()
         for name in os.listdir(active_dir):
             package_path = os.path.realpath(os.path.join(active_dir, name))
+
+            # Work around a long standing bug in realpath which has been open since 2010.
+            # A fix has been proposed since 2013. https://bugs.python.org/issue9949
+            # in windows realpath doesn't resolve symlinks
+            if is_windows:
+                package_path = nt._getfinalpathname(package_path)
```

Comment:
I will spend more time investigating alternative solutions to working around this windows/python problem.

Reply:
It seems that os.readlink will only work if the top-level node is a symbolic link. This is not necessarily the case so will not work. Much investigation was done on this point to come up with this solution. It seems that trawling the internet, this is the solution.  Symbolic links in python on windows are kind of broken. this is the work-around.

Reply:
Can you explain how this code would encounter a file that isn't a symlink? It looks like (and the method's docstring says that) this should only encounter symlinks, since the `active/` dir is expected to contain only symlinks to package directories. If there's an exception to that I'd like to know so I can at least update the docstring here.

If this method really does need to anticipate non-symlink files, and `os.readlink()` only works on symlinks, you could test that the file is a symlink before calling `readlink()` on it. Quick sketch:

```
if os.path.islink(package_path):
    link_target = os.readlink(package_path)
    # If the target is a relative path, convert it to an absolute path.
    if not os.path.isabs(link_target):
        link_target = os.path.normpath(os.path.join(package_path, link_target))
    package_path = link_target
```

`os.path.islink()` detects NTFS symlinks starting with Python 3.2, so that should work on Windows. The advantage here is that we're using only stdlib functions that are intended for external use, making it less likely that the API we're using will change beneath us; and these functions should all work cross-platform, so we woudn't need to have platform checks here.